instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/PeerServiceResolverImpl.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter;
+
+import static java.util.Comparator.comparing;
+import static java.util.Comparator.naturalOrder;
+import static java.util.Comparator.nullsFirst;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+class PeerServiceResolverImpl implements PeerServiceResolver {
+
+  private static final Logger logger = Logger.getLogger(PeerServiceResolverImpl.class.getName());
+
+  private static final Map<String, Map<ServiceMatcher, String>> mapping = new ConcurrentHashMap<>();
```

Comment:
This should not be a static field anymore:
```suggestion
  private final Map<String, Map<ServiceMatcher, String>> mapping = new ConcurrentHashMap<>();
```

You could also make it a normal `HashMap`, since it's not gonna be modified after the `PeerServiceResolver` is built